### PR TITLE
Update DNS settings.

### PR DESCRIPTION
**Pull request by brendandburns**
**What this PR does / why we need it**:
Update kube-dns and kube-dns-masq

**Special notes for your reviewer**:

**Release note**:
```release-note
Update kube-dns to v 1.13.0
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/809)

### CODE REVIEW

File: pkg/acsengine/const.go [hunk 1]
```diff
@@ -57,9 +57,9 @@ var KubeImages = map[api.OrchestratorVersion]map[string]string{
 		"exechealthz":  "exechealthz-amd64:1.2",
 		"addonresizer": "addon-resizer:1.6",
 		"heapster":     "heapster:v1.2.0",
-		"dns":          "kubedns-amd64:1.7",
+		"dns":          "k8s-dns-kube-dns-amd64:1.13.0",
```
Comment (JackQuincy) [inline pkg/acsengine/const.go]: do we want this repeated

Comment (brendandburns) [inline pkg/acsengine/const.go]: Yes, not sure why, but that's the name:

https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/k8s-dns-kube-dns-amd64

Comment (JackQuincy) [inline pkg/acsengine/const.go]: interesting

Comment (amanohar) [inline pkg/acsengine/const.go]: Should this also be updated for Kubernetes 1.6.2, 1.6.0, and prior versions? If not, then update the PR description & commit as to why for archiving/history.

**Comment by brendandburns** [comment]
We could, are we still letting users deploy 1.6.0/2 clusters?


I don't really feel strongly one way or the other...


--brendan

________________________________

**Comment by amanohar** [comment]
No we aren't in both RPs (old and new). Changed for 1.6.6 rolled out for the old RP and the new RP won't allow creation of 1.6.2. But we are not disallowing creates from ACS Engine for those versions.

**Comment by brendandburns** [comment]
I thought I knew how to regenerate the templates, but it seems like I don't...  I thought the scripts/rebase.sh would do it for me...


Pointers?


Thanks

--brendan



________________________________

**Comment by amanohar** [comment]
@brendanburns, you just need to run "make build" (or "make ci" to build & test)

**Comment by brendandburns** [comment]
templates.go should now be fixed...

**Comment by brendandburns** [comment]
Not sure why this is failing. It seems to pass on my machine, but I would expect that I need updates to the testdata files. How do I generate those?

**Comment by amanohar** [comment]
Test data files don't need to be generated anymore. The CI tests are failing: https://jenkins.azure-containers.io/job/acs-engine-pr/642/ (click on details for the link).

Sometime they fail for a variety of reasons like delay in public IP being exposed etc. but I took a quick look at the test run from this PR specifically and they all seem to be failing because of: kube DNS not being up
1) https://jenkins.azure-containers.io/job/acs-engine-pr/642/artifact/_logs/validate-x-kubernetes-westus-642-0-0.log
[06/20/17 03:33:21 UTC] [122 main /gopath/src/github.com/Azure/acs-engine/test/../test/cluster-tests/kubernetes/test.sh] **Checking Kube-DNS**
[06/20/17 03:33:21 UTC] [125 main /gopath/src/github.com/Azure/acs-engine/test/../test/cluster-tests/kubernetes/test.sh]   ... counting down 12
/20/17 03:34:18 UTC] [125 main /gopath/src/github.com/Azure/acs-engine/test/../test/cluster-tests/kubernetes/test.sh]   ... counting down 1
[06/20/17 03:34:23 UTC] [131 main /gopath/src/github.com/Azure/acs-engine/test/../test/cluster-tests/kubernetes/test.sh] **gave up waiting for kube-dns**

2) https://jenkins.azure-containers.io/job/acs-engine-pr/642/artifact/_logs/x-kubernetes-westus-642-0-1.log

++ date '+%D %T %Z'
+ now='06/20/17 03:44:20 UTC'
+ [[ ! -z /gopath/src/github.com/Azure/acs-engine/test/../_logs/validate-x-kubernetes-westus-642-0-1.log ]]
+ echo '[06/20/17 03:44:20 UTC] [131 main /gopath/src/github.com/Azure/acs-engine/test/../test/cluster-tests/kubernetes/test.sh] gave up waiting for kube-dns'
+ tee -a /gopath/src/github.com/Azure/acs-engine/test/../_logs/validate-x-kubernetes-westus-642-0-1.log
[06/20/17 03:44:20 UTC] [131 main /gopath/src/github.com/Azure/acs-engine/test/../test/cluster-tests/kubernetes/test.sh] **gave up waiting for kube-dns**
gave up waiting for kube-dns'

+ exit -1
+ teardown

**Comment by brendandburns** [comment]
thanks, I will double check that the config works to build a running cluster...



________________________________

**Comment by brendandburns** [comment]
Ok, tests are now passing, I had to update the add on manager too.  please take another look.

Thanks!
--brendan

**Pull request merged**